webmails/rainloop/config/nginx-rainloop.conf

```diff
@@ -0,0 +1,38 @@
+server {
+    listen 80 default_server;
+    listen [::]:80 default_server;
+
+    root /var/www/rainloop;
+    
+    # /dev/stdout (Default), <path>, off
+    access_log off;
+
+    # /dev/stderr (Default), <path>, debug, info, notice, warn, error, crit, alert, emerg
+    error_log /dev/stderr error;
+
+    index index.php;
+
+    location / {
+        try_files $uri $uri/ /index.php?$query_string;
+    }
+
+    location ~ \.php$ {
+        fastcgi_split_path_info ^(.+\.php)(/.*)$;
+        
+        fastcgi_intercept_errors on;
+        fastcgi_index  index.php;
+
```

Comment:
fastcgi_param REMOTE_ADDR $http_x_real_ip; 

Should have something of the sort, no? What about the protocol so that it knows whether it's behind HTTP/HTTPS and sets the right cookie flags?

Reply:
Well, I've tested with two reverse proxies: First one is my [Swag reverse proxy](https://docs.linuxserver.io/general/swag) pointing to port 80 and 443 and handles SSL certificates. The front container running nginx forwards webmail traffic to Rainloop at HTTP port 80. That's no difference with previous Apache setup.

With and without `fastcgi_param REMOTE_ADDR $http_x_real_ip;` works in my setup. 

I can add this line when we're sure.

Reply:
No need to add `fastcgi_param REMOTE_ADDR $http_x_real_ip;` to `nginx-rainloop.conf`. Evidence is shown by running `nginx -T` command (test configuration, dump it and exit) in the rainloop container which shows all enabled settings:

```
...
fastcgi_param  REMOTE_ADDR        $remote_addr;
fastcgi_param  REMOTE_PORT        $remote_port;
fastcgi_param  SERVER_ADDR        $server_addr;
fastcgi_param  SERVER_PORT        $server_port;
fastcgi_param  SERVER_NAME        $server_name;

# PHP only, required if PHP was built with --enable-force-cgi-redirect
fastcgi_param  REDIRECT_STATUS    200;
```

Configuration file stack:
```
/etc/nginx/http.d/rainloop.conf
    include /etc/nginx/fastcgi_params; 
         fastcgi_param  REMOTE_ADDR        $remote_addr;
```